core/primitives/src/runtime/config_store.rs

```diff
@@ -134,6 +132,26 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_data_roundtrip_is_more_expensive() {
+        // We have an assumption that the deepest receipts we can create is by creating recursive
+        // function call promises (calling function call from a function call).
+        // If the cost of a data receipt is cheaper than the cost of a function call, then it's
+        // possible to create a promise with a dependency which will be executed in two blocks that
+        // is cheaper than just two recursive function calls.
+        // That's why we need to enforce that the cost of the data receipt is not less than a
+        // function call. Otherwise we'd have to modify the way we compute the maximum depth.
+        let store = RuntimeConfigStore::new(None);
+        for config in store.store.values() {
```

Comment:
Good catch! That's a good example of specific issue you can get from a rather amorphous problem of "having default doesn't feel right".